web/src/markdown.ts

```diff
@@ -659,51 +799,77 @@ export function parse({raw_content, helper_config}) {
         preprocessors: [preprocess_code_blocks, preprocess_translate_emoticons],
     };
 
-    return parse_with_options({raw_content, helper_config, options});
+    return parse_with_options(raw_content, helper_config, options);
 }
 
+export type ParseOptions = {
+    get_linkifier_regexes: () => RegExp[];
+    linkifierHandler: (pattern: RegExp, matches: LinkifierMatch[]) => string;
+    emojiHandler: (emoji_name: string) => string;
+    unicodeEmojiHandler: (unicode_emoji: string) => string;
+    streamHandler: (stream_name: string) => string | undefined;
+    streamTopicHandler: (stream_name: string, topic: string) => string | undefined;
+    texHandler: (tex: string, fullmatch: string) => string;
+    timestampHandler: (time: string | number) => string;
+    gfm: boolean;
+    tables: boolean;
+    breaks: boolean;
+    pedantic: boolean;
+    sanitize: boolean;
+    smartLists: boolean;
+    smartypants: boolean;
+    zulip: boolean;
+    renderer: marked.Renderer;
+    preprocessors: ((src: string) => string)[];
+};
+
 // NOTE: Everything below this line is likely to be web-specific
 //       and won't be used by future platforms such as mobile.
 //       We may eventually move this code to a new file, but we want
 //       to wait till the dust settles a bit on some other changes first.
 
-let web_app_helpers;
+let web_app_helpers: MarkdownHelpers | undefined;
 
-export function initialize(helper_config) {
+export function initialize(helper_config: MarkdownHelpers): void {
     // This is generally only intended to be called by the web app. Most
     // other platforms should call setup().
     web_app_helpers = helper_config;
 }
 
-export function apply_markdown(message) {
+export function apply_markdown(message: {raw_content: string}): {
```

Comment:
[prep commit TODO] converting this to a function that returns a message instead of modifying it. Probably it can take `raw_content` as its parameter instead.